File: baek_joon/cpp_src/step5/10818.cpp
#include <iostream>
using namespace std;

int main()
{
    int N, min, max;
    int* array = new int;

    cin >> N;

    // int array[N]; C++17

    for (int i = 0; i < N; i++) 
    {
        cin >> array[i];
    }
    
    min = array[0];
    max = array[0];

    for (int i = 1; i < N; i++)
    {
        if (max < array[i])
        {
            max = array[i];
        }

        if (min > array[i])
        {
            min = array[i];
        }
    }
    cout << min << " " << max;
}

File: baekjoon/rust/10818.rs
use std::io::{read_to_string, stdin};

fn main() {
    let buf = read_to_string(stdin()).unwrap();
    let mut inputs = buf.trim().lines();

    inputs.next().unwrap();

    let nums: Vec<i32> = inputs
        .next()
        .unwrap()
        .split_whitespace()
        .map(|v| v.parse().unwrap())
        .collect();

    let min_num = nums.iter().min().unwrap();
    let max_num = nums.iter().max().unwrap();

    print!("{} {}", *min_num, *max_num);
}

fn main() {
    let buf = read_to_string(stdin()).unwrap();

    const N: i32 = 1_000_000;
    let (mut min, mut max) = (N, -N);

    buf.split_whitespace()
        .skip(1)
        .map(|s| s.parse::<i32>().unwrap())
        .for_each(|num| {
            min = num.min(min);
            max = num.max(max);
        });

    println!("{min} {max}");
}
